torch/onnx/symbolic_opset9.py

```diff
@@ -3865,7 +3865,9 @@ def zeros_like(
 ):
     shape = g.op("Shape", input)
     if dtype is None:
-        scalar_type = _type_utils.JitScalarType.FLOAT
+        scalar_type = symbolic_helper._try_get_scalar_type(input)
+        if scalar_type is None:
+            scalar_type = _type_utils.JitScalarType.FLOAT
```

Comment:
```suggestion
        scalar_type = _type_utils.JitScalarType.from_value(
            arg, _type_utils.JitScalarType.FLOAT
        )
```

Reply:
A more efficient single liner is using `JitScalarType.from_value()` directly, which accepts a default `scalar_type` when it can't get it from `input`. It prevents unnecessary conditionals and simplify the code

Same applies for occurrences below

Reply:
This is much better. I updated all the changes to use `JitScalarType.from_value()`. Thank you!